tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.cc

```diff
@@ -75,6 +78,25 @@ static void ParseGCSPath(absl::string_view fname, bool object_empty_ok,
   strcpy(*object, object_view.data());
 }
 
+class TempFile : public std::fstream {
+ public:
+  // We should specify openmode each time we call TempFile.
+  TempFile(const char* temp_file_name, std::ios::openmode mode)
+      : std::fstream(temp_file_name, mode), name(temp_file_name) {}
+  TempFile(TempFile&& rhs) : std::fstream(std::move(rhs)), name(rhs.name) {
+    rhs.name = nullptr;
+  }
+  ~TempFile() {
+    std::fstream::close();
+    std::remove(name);
+    plugin_memory_free(const_cast<char*>(name));
+  }
+  const char* getName() { return name; }
+
+ private:
+  const char* name;
+};
+
```

Comment:
We should move this to a separate file to keep this one only containing the API glue

Reply:
Done